Suggested changes to README (#1)
valory-xyz/open-operator-aks: README.md

After the deployment process finishes, the agent will be running in a [`screen`](https://www.gnu.org/software/screen/) session within an [AWS EC2](https://aws.amazon.com/ec2/) instance in the default AWS Region `us-east-2`.[^1]

> **Note** <br />
> **This repository contains default configuration parameters in the `./config/service_vars.env` file that you will need to override with the information you will have been provided by the service owner.**

> [^1]: If you wish to deploy on another AWS Region, you need to modify the Terraform variable `deployment_region` in the file `./infra/aws/docker_compose/variables.tf`. You also need to provide a valid Amazon Machine Image (AMI) ID for that region (resource `aws_instance` in the file `./infra/aws/docker_compose/main.tf`), otherwise the deployment process will fail on `terraform apply`.

Reviewer: This is a footnote (will be rendered at the bottom of the page). Please keep it in the previous position (closer to where the footnote is referenced [^1])

Author: OK

Author: Actually it renders exactly the same as the original, so no change is needed AFAICT

## Prerequisites

1. **Set up your AWS account.** Sign in to the AWS Management Console and configure the following parameters.

>    1. In case you don't have one, you need to create an IAM user with an access key. Within the AWS Management Console, create a new user (IAM/Users), and [create an access key](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_access-keys.html) for that user (Security credentials/Access keys). The IAM user needs have the permissions `AmazonEC2FullAccess`, `AmazonS3FullAccess` and `IAMFullAccess`. Note down the *AWS Access Key ID* and *AWS Secret Access Key*.

Reviewer: I think IAMFullAccess is not required, but AmazonVPCFullAccess maybe is required.

Reviewer: I have just tested it. you only need `AmazonEC2FullAccess` and `AmazonS3FullAccess`

  
   2. You also need to [create an *S3 bucket*](https://docs.aws.amazon.com/AmazonS3/latest/userguide/create-bucket-overview.html) to store the Terraform state for the service.[^2] Create the bucket in region `us-east-2`.  You must follow the [AWS guidelines](https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucketnamingrules.html) for naming your bucket. Note down the bucket name.

   [^2]: For simplicity, the Terraform scripts in this repository do not implement [state locking](https://developer.hashicorp.com/terraform/language/state/locking). Therefore, it is important to ensure that the script is not executed concurrently by different users in order to prevent potential issues. You might consider implementing state locking in the AWS S3 bucket using [DyanomDB](https://aws.amazon.com/dynamodb/). See for example [this](https://terraformguru.com/terraform-real-world-on-aws-ec2/20-Remote-State-Storage-with-AWS-S3-and-DynamoDB/) or [this](https://blog.gruntwork.io/how-to-manage-terraform-state-28f5697e68fa) tutorial.

2. **Prepare an SSH key pair.** This key pair will be used to access the deployed AWS EC2 instance where the service will be running.

>    You can generate the key pair with the following command and _do not enter a password_ when prompted to do so

Reviewer: ```suggestion
   You can generate the key pair with the following command. **Do not enter a passphrase if prompted to do so:**
```

   ```bash

>    ssh-keygen -t rsa -b 2048 -f id_rsa

Reviewer: This creates a key without prompting a passphrase:

```suggestion
   ssh-keygen -t rsa -b 2048 -N "" -f id_rsa
```

Author: Nice!

   ```

>    Store securely both the public and private key.

Reviewer: Why remove this information? Put it as a footnote at least?

Author: ok

1. **Fork this repository.**
2. **Configure cloud credentials and service parameters.**

>    First, define the following [secrets](https://docs.github.com/en/actions/security-guides/encrypted-secrets#creating-encrypted-secrets-for-a-repository) and [variables](https://docs.github.com/en/actions/learn-github-actions/variables#creating-configuration-variables-for-a-repository) in the forked repository, we recommeng using Github's web UI:

Reviewer: ```suggestion
   First, [define the following secrets](https://docs.github.com/en/actions/security-guides/encrypted-secrets#creating-encrypted-secrets-for-a-repository) and [variables](https://docs.github.com/en/actions/learn-github-actions/variables#creating-configuration-variables-for-a-repository) in the forked repository. We recommend defining them through the GitHub's website UI:
```

   ```

   > **Warning** <br />

>    > **In general the `keys.json` file should not be exposed in the repository. The recommended practice is to [define the GitHub secret](https://docs.github.com/en/actions/security-guides/encrypted-secrets#creating-encrypted-secrets-for-a-repository) `KEYS_JSON` with the contents of the file.**

Reviewer: ```suggestion
   > **Typically, it is not advisable to expose the `keys.json` file in the repository. As an alternative, we provide a mechanism where you can [define the GitHub secret](https://docs.github.com/en/actions/security-guides/encrypted-secrets#creating-encrypted-secrets-for-a-repository) `KEYS_JSON` with the contents of the file.**
```

   ```

   > **Warning** <br />

>    > **If you don't want to expose secret/confidential variables in the repository, you can assign them a blank value (or a placeholder value)[^3] in the file `service_vars.env`, and override their values by [defining GitHub secrets](https://docs.github.com/en/actions/security-guides/encrypted-secrets#creating-encrypted-secrets-for-a-repository) matching the corresponding variables' names. In the present case, we recommend this practice with <code>ARBITRUM_RPC_0</code> and <code>SERVICE_RPC_0</code>.**

Reviewer: ```suggestion
   > **If you don't want to expose secret/confidential variables in the repository, you can assign them a blank value (or a placeholder value)[^3] in the file `service_vars.env`, and override their values by [defining GitHub secrets](https://docs.github.com/en/actions/security-guides/encrypted-secrets#creating-encrypted-secrets-for-a-repository) matching the corresponding variables' names. For the present case (`keep3r_bot` and `keep3r_bot_goerli` services), we recommend this practice with the variables <code>ARBITRUM_RPC_0</code> and <code>SERVICE_RPC_0</code>.**
```

   <p align="center"><img src="docs/images/deploy-service-workflow.png" alt="Deploy service workflow" width=80%></p>

>    You should see the following output in the *Summary* step of the workflow:

Reviewer: Does really removing this helps? How will the user know the IP address to connect otherwise? It's not explained anywhere else in the repo.
